packages/react-intl/src/formatters/message.ts

```diff
@@ -72,6 +80,24 @@ function deepMergeFormatsAndSetTimeZone(
   };
 }
 
+export function patchFormatXMLElementFn(
+  values: FormatMessageValues
+): FormatMessageValues {
+  return Object.keys(values).reduce((acc: FormatMessageValues, k) => {
+    const v = values[k];
+    acc[k] =
+      typeof v === 'function'
+        ? unapplyFormatXMLElementFn(
+            // type assertation needed since TypeScript infers possible generic
+            // Function type for `v`, but we know the only possible function
+            // in `values` is FormatXMLElementFn
+            v as FormatXMLElementFn<React.ReactNode, React.ReactNode>
+          )
+        : v;
+    return acc;
+  }, {});
+}
+
```

Comment:
Iterate over `values` and patch all `functions` found. Is not very safe, but AFAIK there is no other way to distinguish `FormatXMLElementFn` and currently, there cannot be passed other functions than `FormatXMLElementFn`.